### PR TITLE
Fix the silent error in `tm_a_pca`

### LINKED ISSUE #644: selecting color variable on biplot causes silent failure
Open app, switch to biplot, select any color variable, output becomes empty.

```
# CDISC data example

library(teal.widgets)

data <- teal_data()
data <- within(data, {
  library(nestcolor)
  ADSL <- rADSL
})
datanames <- c("ADSL")
datanames(data) <- datanames
join_keys(data) <- default_cdisc_join_keys[datanames]

app <- init(
  data = data,
  modules = modules(
    teal.modules.general::tm_a_pca(
      "PCA",
      dat = data_extract_spec(
        dataname = "ADSL",
        select = select_spec(
          choices = variable_choices(
            data = data[["ADSL"]], c("BMRKR1", "AGE", "EOSDY")
          ),
          selected = c("BMRKR1", "AGE"),
          multiple = TRUE
        ),
        filter = NULL
      ),
      ggplot2_args = ggplot2_args(
        labs = list(subtitle = "Plot generated by PCA Module")
      )
    )
  )
)

if (interactive()) {
  shinyApp(app$ui, app$server)
}
```

image

### PR DESCRIPTION
**Pull request by vedhav**
Closes #644

### CODE REVIEW

File: R/tm_a_pca.R [hunk 1]
```diff
@@ -294,11 +294,10 @@ srv_a_pca <- function(id, data, reporter, filter_panel_api, dat, plot_height, pl
...
       response[[i]]$select$selected <- NULL
-      response[[i]]$select$choices <- var_labels(isolate(data())[[response[[i]]$dataname]])
-      response[[i]]$select$choices <- setdiff(
-        response[[i]]$select$choices,
-        unlist(teal.data::join_keys(isolate(data()))[[response[[i]]$dataname]])
-      )
+      all_cols <- var_labels(isolate(data())[[response[[i]]$dataname]])
+      ignore_cols <- unlist(teal.data::join_keys(isolate(data()))[[response[[i]]$dataname]])
+      color_cols <- all_cols[!names(all_cols) %in% ignore_cols]
+      response[[i]]$select$choices <- choices_labeled(names(color_cols), color_cols)
```
Comment (gogonzo) [inline R/tm_a_pca.R:300]:
```suggestion
      all_cols <- colnames(isolate(data())[[response[[i]]$dataname]])
      ignore_cols <- unlist(teal.data::join_keys(isolate(data()))[[response[[i]]$dataname]])
      color_cols <- all_cols[!all_cols %in% ignore_cols]
      response[[i]]$select$choices <- choices_labeled(color_cols, color_cols)
```

Comment (gogonzo) [reply R/tm_a_pca.R:300]: @vedhav sorry this is not  good suggestion. But I have a feeling that it can be still simplified

Comment (vedhav) [reply R/tm_a_pca.R:300]: I just made use of the var labels. Please pull the branch, as these labels are retained.

Comment (gogonzo) [reply R/tm_a_pca.R:300]: Cool, I think you can replace `var_labels` with `teal.data::col_labels`

Comment (vedhav) [reply R/tm_a_pca.R]: Thanks! Changed it. Only two other places use `var_labels`. Let me check if those can also be replaced so we can remove the internal function completely.

**Review by gogonzo — APPROVED** [review]

**Pull request merged**